### PR TITLE
add fedtoken contract

### PR DESCRIPTION
**Pull request by btree1970**
Add the base for governence contract

### CODE REVIEW

File: test/FedToken.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  describe("Deployment", function () {
+      
+    it("Has the right initialization", async function () {
+      expect(await FEDToken.symbol()).to.equal("FED");
+      expect(await FEDToken.name()).to.equal("FedToken")
+    });
+
+    it("Owner can mint", async function () {
+        expect(await FEDToken.totalSupply()).to.equal(0)
+        await FEDToken.mint(owner.address, 100000)
```
Comment (ksgebrekirstos98) [inline test/FedToken.js:52]: Where are these variables defined?

- totalSupply
- mint

File: contracts/governence/FedToken.sol [hunk 2]
```diff
@@ -0,0 +1,18 @@
+//SPDX-License-Identifier: UNLICENSED
+
+// Solidity files have to start with this pragma.
+// It will be used by the Solidity compiler to validate its version.
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetMinterPauser.sol";
+
+
+contract FedToken is ERC20PresetMinterPauser {
```
Comment (ksgebrekirstos98) [inline contracts/governence/FedToken.sol:10]: Could you add some comments about what this contract is and what it's supposed to do?

File: test/FedToken.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  // Mocha has four functions that let you hook into the the test runner's
+  // lifecyle. These are: `before`, `beforeEach`, `after`, `afterEach`.
+
+  // They're very useful to setup the environment for tests, and to clean it
+  // up after they run.
+
+  // A common pattern is to declare some variables, and assign them in the
+  // `before` and `beforeEach` callbacks.
+
+  let Token;
```
Comment (ksgebrekirstos98) [inline test/FedToken.js:18]: nit: you can delete lines 18, 21, 22 & 23 if you're not using them

File: test/FedToken.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    it("Owner can mint", async function () {
+        expect(await FEDToken.totalSupply()).to.equal(0)
+        await FEDToken.mint(owner.address, 100000)
+        expect(await FEDToken.totalSupply()).to.equal(100000)
+        expect(await FEDToken.balanceOf(owner.address)).to.equal(100000)
+    });
+
+
+    // it("Should assign the total supply of tokens to the owner", async function () {
```
Comment (ksgebrekirstos98) [inline test/FedToken.js:58]: Can you remove this commented code?

File: test/FedToken.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+
+// Hardhat tests are normally written with Mocha and Chai.
+
+// We import Chai to use its asserting functions here.
+const { expect } = require("chai");
+const { ethers } = require("hardhat")
+
+describe("FedToken contract", function () {
+  // Mocha has four functions that let you hook into the the test runner's
```
Comment (ksgebrekirstos98) [inline test/FedToken.js:9]: Unnecessary cruft leftover from Hardhat. We can remove all the comments that aren't relevant to this class

**Review by ksgebrekirstos98 — APPROVED** [review]

File: test/FedToken.js (hunk 4 above)
Comment (btree1970) [reply test/FedToken.js:58]: sounds good

File: test/FedToken.js (hunk 3 above)
Comment (btree1970) [reply test/FedToken.js:18]: sounds good

File: test/FedToken.js (hunk 5 above)
Comment (btree1970) [reply test/FedToken.js:9]: sounds good

File: contracts/governence/FedToken.sol (hunk 2 above)
Comment (btree1970) [reply contracts/governence/FedToken.sol:10]: will had in the future

File: test/FedToken.js (hunk 1 above)
Comment (btree1970) [reply test/FedToken.js:52]: check the inheritance tree

**Pull request merged**